client/src/areas/grail/GrailArea.tsx

```diff
@@ -173,20 +185,36 @@ const GrailAreaInternal: FC<Props> = props => {
 
   function setThemeAndTitle() {
     let theme: IAppTheme = null;
+
+    let defaultThemeToSet = defaultTheme;
+    let ethThemeToSet = ethTheme;
+    let runewordThemeToSet = runewordTheme;
+
+    if (darkThemeIsEnabled()) {
+      defaultThemeToSet = defaultThemeDark;
+      ethThemeToSet = ethThemeDark;
+      runewordThemeToSet = runewordThemeDark;
+    }
+
     switch (grailMode) {
       case GrailMode.Eth:
-        theme = ethTheme;
+        theme = ethThemeToSet;
         break;
       case GrailMode.Runeword:
-        theme = runewordTheme;
+        theme = runewordThemeToSet;
         break;
       default:
-        break;
+        theme = defaultThemeToSet;
     }
 
     setAppTheme(theme);
   }
 
+  function toggleDarkTheme() {
+    toggleDarkThemeEnabled();
```

Comment:
get rid of this function and your global variables.
instead save or remove from the local storage if it should currently be in dark mode.

Then edit the `App.tsx` and read the initial value of the local storage setting... 
it should look like this (not working code):

```
export const App: FC = () => {
  const [appTheme, setAppTheme] = useState<IAppTheme>(geTheme(defaultTheme, localeStorage.getValue("darkThemeSetting"));

  return (
    <AppThemeContext.Provider value={{ appTheme, setAppTheme: handleSetTheme }}>

```

I hope this gives you an idea how to do it. Else just ask, I can also edit your PR myself if you want.